crm_lead_address_street3/tests/test_crm_lead_address_street3.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+import odoo
+
+from .util.odoo_tests import TestBase
+from .util.singleton import Singleton
+
+log = logging.getLogger('Test')
+
+
+class TestMemory(object):
+    """Keep records in memory across tests."""
+    __metaclass__ = Singleton
+
+
+@odoo.tests.common.at_install(False)
+@odoo.tests.common.post_install(True)
+class Test(TestBase):
+
+    def setUp(self):
+        super(Test, self).setUp()
+        self.memory = TestMemory()
+
+    def test_0001_create_lead(self):
+
+        log.info("""
+            - In order to test that the street3 is well passed to the future
+            partner, I create a new lead.
+        """)
+
+        lead, = self.createAndTest(
```

Comment:
Create the lead in `setUp`, with just `self.env["crm.lead"].create(...)`. No need for all this boilerplate.